src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import * as i18n from '@/languages';
+import Animated, { useDerivedValue } from 'react-native-reanimated';
+import { AnimatedText, Box, Inline, Text, TextIcon, useForegroundColor } from '@/design-system';
+import { opacity } from '@/__swaps__/utils/swaps';
+import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
+import { SwapPriceImpactType } from '@/hooks/usePriceImpactDetails';
+
+export const PriceImpactWarning = () => {
+  const { AnimatedSwapStyles, PriceImpactWarning } = useSwapContext();
+
+  // TODO: i18n is not liking this...
+  const warningText = useDerivedValue(() => {
+    if (PriceImpactWarning.value.type === SwapPriceImpactType.none) return '';
+    return i18n.t(i18n.l.exchange.price_impact.you_are_losing, {
+      impactDisplay: PriceImpactWarning.value.impactDisplay,
+    });
+  });
```

Comment:
is there a way to use i18n inside of a derived value here? @christianbaroni

Reply:
<img width="585" alt="Screenshot 2024-04-17 at 1 14 27 PM" src="https://github.com/rainbow-me/rainbow/assets/6284525/23b954ac-fa3d-4ca8-b7f5-f4c84ac50f2e">